apple/internal/transition_support.bzl

```diff
@@ -256,10 +256,10 @@ def _xcframework_split_attr_key(*, cpu, environment, platform_type):
         platform_type = platform_type,
     ) + "_" + environment
 
-def _resolved_cpu_for_cpu(*, cpu, environment):
+def _resolved_cpu_for_cpu(*, cpu, environment, platform_type):
     # TODO(b/180572694): Remove cpu redirection after supporting platforms based toolchain
     # resolution.
-    if cpu == "arm64" and environment == "simulator":
+    if cpu == "arm64" and environment == "simulator" and platform_type != "watchos":
```

Comment:
want to link https://github.com/bazelbuild/bazel/pull/16181 ?

Reply:
Yeah, since that will require this to be reverted.